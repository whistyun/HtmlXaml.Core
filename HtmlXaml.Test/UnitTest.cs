
using ApprovalTests;
using ApprovalTests.Reporters;
using HtmlXaml.Core;
using NUnit.Framework;
using System.Linq;
using System.Threading;
using System.Windows.Documents;

namespace HtmlXaml.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest
    {
        [Test]
        [Apartment(ApartmentState.STA)]
        public void Button()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void CodeBlock()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void InlineCode()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Input()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void List()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Progres()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void TypicalBlock()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void TypicalInline()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
