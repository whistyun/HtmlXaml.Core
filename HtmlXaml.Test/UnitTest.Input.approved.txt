<?xml version="1.0" encoding="utf-16"?>
<FlowDocument xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
  <Paragraph>
    <TextBox IsReadOnly="True" Width="100" xml:space="preserve"></TextBox> <TextBox IsReadOnly="True" Width="98" xml:space="preserve"></TextBox> <TextBox IsReadOnly="True" Width="100" xml:space="preserve"></TextBox> <TextBox IsReadOnly="True">textbox</TextBox></Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <Button Width="100" IsEnabled="False" xml:space="preserve"></Button> <Button Width="98" IsEnabled="False" xml:space="preserve"></Button> <Button Width="100" IsEnabled="False" xml:space="preserve"></Button> <Button IsEnabled="False">button</Button></Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <Button Width="100" IsEnabled="False" xml:space="preserve"></Button> <Button Width="98" IsEnabled="False" xml:space="preserve"></Button> <Button Width="100" IsEnabled="False" xml:space="preserve"></Button> <Button IsEnabled="False">button</Button></Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <Button Width="100" IsEnabled="False" xml:space="preserve"></Button> <Button Width="98" IsEnabled="False" xml:space="preserve"></Button> <Button Width="100" IsEnabled="False" xml:space="preserve"></Button> <Button IsEnabled="False">button</Button></Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <RadioButton IsEnabled="False" /> <RadioButton IsChecked="True" IsEnabled="False" /> <RadioButton IsChecked="True" IsEnabled="False" /></Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <CheckBox IsEnabled="False" /> <CheckBox IsChecked="True" IsEnabled="False" /> <CheckBox IsChecked="True" IsEnabled="False" /></Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <Slider Minimum="0" Maximum="100" Value="50" Width="100" IsEnabled="False" /> <Slider Minimum="0" Maximum="200" Value="50" Width="100" IsEnabled="False" /> <Slider Minimum="50" Maximum="100" Value="90" Width="100" IsEnabled="False" /></Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <TextBox TextWrapping="Wrap" AcceptsReturn="True" AcceptsTab="True" xml:space="preserve">asasdfasdfsadfasdfasd

sd

df</TextBox>
  </Paragraph>
  <BlockUIContainer Tag="RuleSingle">
    <Separator />
  </BlockUIContainer>
  <Paragraph>
    <TextBox TextWrapping="Wrap" AcceptsReturn="True" AcceptsTab="True" Width="210" Height="42" xml:space="preserve">afsadfasdfasd

sd

df</TextBox>
  </Paragraph>
</FlowDocument>
